### PR TITLE
Set Java 23 EE compliance accordingly

### PR DESCRIPTION
**Pull request by akurtakov**
## What it does
Sets Java 23 EE compliance to Java 23 now that constants are in jdt.core via https://github.com/eclipse-jdt/eclipse.jdt.core/pull/2345. 
Fixes https://github.com/eclipse-pde/eclipse.pde/issues/1227

## How to test
org.eclipse.pde.core.tests.internal.classpath.ClasspathResolutionTest should succeed with this patch.

## Author checklist

- [x] I have thoroughly tested my changes
- [x] The change is following the [coding conventions](https://wiki.eclipse.org/Platform/How_to_Contribute#Coding_Conventions)
- [x] I have signed the [Eclipse Contributor Agreement (ECA)](https://www.eclipse.org/legal/ECA.php)

### CODE REVIEW

**Comment by akurtakov** [comment]
@iloveeclipse Verification here will fail till there is I-build with the new jdt.core constants. I leave it to you whether to push this one now and have the fix sooner or wait for verification to succeed.

**Review by iloveeclipse — APPROVED** [review]
Will merge once all other PRs are through for the unstable build.

**Pull request merged**
